app/javascript/react/containers/ShowLocationContainer.js

```diff
@@ -54,11 +55,27 @@ class ShowLocationContainer extends React.Component {
   componentDidMount(){
     fetch(`/api/v1/locations/${this.props.match.params.id}`)
     .then(response => response.json())
-    .then(response =>{
+    .then(response => {
       this.setState({chosenLocation: response.location, reviews: response.reviews})
     })
   }
 
+  handleDeleteLocation(event) {
+    event.preventDefault()
+    let locationId = this.state.chosenLocation.id
+    fetch(`/api/v1/locations/${locationId}`, {
+      credentials: 'same-origin',
+      method: 'DELETE',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+    .then(response => response.json())
+    .then(remainingLocations => {
+      this.props.history.push(`/locations`, { locations: remainingLocations } )
+    })
+  }
```

Comment:
Ill allow this, but it is better practice to redirect using the Redirect component

https://reacttraining.com/react-router/web/api/Redirect